gluten-core/src/main/scala/org/apache/gluten/utils/PullOutProjectHelper.scala

```diff
@@ -70,6 +76,19 @@ trait PullOutProjectHelper {
           .toAttribute
     }
 
+  protected def getMissingLiterals(exprs: Seq[NamedExpression]): Seq[NamedExpression] = {
+    val literalMap = new mutable.HashMap[Expression, NamedExpression]()
+    exprs.filter {
+      case alias: Alias =>
+        alias.child match {
+          case _: Literal =>
+            literalMap.put(alias.child.canonicalized, alias).nonEmpty
```

Comment:
Can you add a comment here explaining that the purpose of the HashMap is to obtain duplicate literals?